tests/rptest/tests/transactions_test.py

```diff
@@ -0,0 +1,238 @@
+        while num_consumed_records != self.max_records:
+            self.check_consumer(consumer1)
+
+            producer.begin_transaction()
+
+            for record in self.records:
+                assert (record.error() == None)
+                producer.produce(self.output_t.name, record.value(),
+                                 record.key(), 0)
+
+            producer.send_offsets_to_transaction(
+                consumer1.position(consumer1.assignment()),
+                consumer1.consumer_group_metadata())
+
+            producer.commit_transaction()
+
+            num_consumed_records += len(self.records)
+
+        producer.flush()
+        consumer1.close()
+
+        consumer2 = ck.Consumer({
+            'group.id': "testtest",
+            'bootstrap.servers': self.redpanda.brokers(),
+            'auto.offset.reset': 'earliest',
+        })
+        consumer2.subscribe([self.output_t])
+
+        final_consume_cnt = 0
+        expected = bytes("0", 'UTF-8')
+
+        while final_consume_cnt != self.max_records:
+            self.check_consumer(consumer2)
+
+            for record in self.records:
+                assert (record.key() == expected)
+                assert (record.value() == expected)
+                expected = bytes(str(int(record.key()) + 1), 'UTF-8')
+            final_consume_cnt += len(self.records)
+
+    @cluster(num_nodes=3)
+    def rejoin_member_test(self):
+        self.generate_data(self.input_t, self.max_records)
+
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': '0',
+            'transaction.timeout.ms': 10000,
+        })
+
+        group_name = "test"
+        consumer1 = ck.Consumer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'group.id': group_name,
+            'auto.offset.reset': 'earliest',
+            'enable.auto.commit': False,
+        })
+
+        producer.init_transactions()
+
+        consumer1.subscribe([self.input_t])
+        self.check_consumer(consumer1)
+
+        producer.begin_transaction()
+
+        for record in self.records:
+            assert (record.error() == None)
+            producer.produce(self.output_t.name, record.value(), record.key(),
+                             0)
+
+        offsets = consumer1.position(consumer1.assignment())
+        metadata = consumer1.consumer_group_metadata()
+
+        consumer2 = ck.Consumer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'group.id': group_name,
+            'auto.offset.reset': 'earliest',
+            'enable.auto.commit': False,
+        })
+
+        consumer2.subscribe([self.input_t])
+        # Rejoin can take some time, so we chould swithc off consume timeout
+        self.check_consumer(consumer2, False)
+
+        try:
+            producer.send_offsets_to_transaction(offsets, metadata, 2)
+            assert False and "send_offsetes should fail"
```

Comment:
`assert False, "send_offsetes should fail"` not `assert False and "send_offsetes should fail"`